akka-actor/src/main/scala/akka/routing/ConsistentHashing.scala

```diff
@@ -149,7 +149,10 @@ final case class ConsistentHashingRoutingLogic(
   def this(system: ActorSystem) =
     this(system, virtualNodesFactor = 0, hashMapping = ConsistentHashingRouter.emptyConsistentHashMapping)
 
-  private val selfAddress = system.asInstanceOf[ExtendedActorSystem].provider.getDefaultAddress
+  // important that this is lazy, because consistent hashing routing pool is used by SimpleDnsManager
+  // that can be activated early, before the transport defaultAddress is set in the startup
+  // see issue #20263
+  private lazy val selfAddress = system.asInstanceOf[ExtendedActorSystem].provider.getDefaultAddress
```

Comment:
You are right, we can do better. I have changed it slightly so that if it happens the message will not be routed, but it will try again for next message, see https://github.com/akka/akka/pull/20264/files#diff-0931c33dd37d5c175c80c0b6244f3d95R162